ui/admin/app/controllers/scopes/scope/workers/worker/tags.js

```diff
@@ -4,10 +4,73 @@
  */
 
 import Controller, { inject as controller } from '@ember/controller';
-import { TAG_TYPE_CONFIG } from 'api/models/worker';
+import { inject as service } from '@ember/service';
+import { tracked } from '@glimmer/tracking';
+import { action } from '@ember/object';
+import { loading } from 'ember-loading';
+import { notifySuccess, notifyError } from 'core/decorators/notify';
+import { TAG_TYPE_CONFIG, TAG_TYPE_API } from 'api/models/worker';
 
 export default class ScopesScopeWorkersWorkerTagsController extends Controller {
   @controller('scopes/scope/workers/index') workers;
 
+  // =services
+  @service intl;
+
+  // =attributes
+
   tagTypeConfig = TAG_TYPE_CONFIG;
+  tagTypeApi = TAG_TYPE_API;
+
+  @tracked removeModal = false;
+  @tracked removalConfirmation = null;
+  @tracked modalTag = null;
+
+  /**
+   * Build the display name for a worker tag
+   * @type {string}
+   */
+  get tagDisplayName() {
+    return `"${this.modalTag?.key} = ${this.modalTag?.value}"`;
+  }
+
+  /**
+   * Checks if user entered REMOVE to enable the remove button.
+   * @type {boolean}
+   */
+  get isRemovalConfirmed() {
+    return (
+      this.removalConfirmation === this.intl.t('actions.remove').toUpperCase()
+    );
+  }
+
+  /**
+   * Removes selected tag from the worker.
+   */
+  @action
+  @loading
+  @notifyError(({ message }) => message, { catch: true })
+  @notifySuccess('resources.worker.tags.remove.success')
+  async removeApiTag() {
+    await this.model.removeApiTags({
+      [this.modalTag.key]: [this.modalTag.value],
+    });
+    this.removeModal = false;
+    this.modalTag = null;
+    this.removalConfirmation = null;
+  }
+
+  /**
+   * Opens the remove tag modal.
+   * @param {string} action
+   * @param {object} apiTag
+   */
+  @action
+  toggleModal(action, apiTag = null) {
+    if (action === 'remove') {
```

Comment:
What's the purpose of this `action`? Are you trying to share this with edit? Why not just use two separate functions instead?

Reply:
Yes the plan was to use `toggleModal` for both as they do the same thing (except one toggles remove and one toggles edit) but happy to split them up.

Reply:
Agreed with Zed, if this is to share `remove` and `edit` actions, I will create two separate methods for those, that deal with the edit or removal state of the tags, and rely on `toggleModal` to deal with the modal state.

Reply:
Updated so it is specific to just the remove modal.